sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -149,7 +145,7 @@ public class ServiceBusReceiver : IAsyncDisposable
                     sessionId: sessionId,
                     isSessionReceiver: IsSessionReceiver);
                 _scopeFactory = new EntityScopeFactory(EntityPath, FullyQualifiedNamespace);
-                _plugins = plugins;
+                _pipeline = new ServiceBusPipeline(plugins.ToArray());
```

Comment:
Should we consider taking this in as an array instead of list?   If our parameter declares the form that we need, it provides the caller insight and the caller may be able to cache the transform so that it doesn't happen for each client that is spawned.

Reply:
Woops, you are right. The set of plugins is immutable on client creation now so we should do the conversion there.